keys/keybase.go

```diff
@@ -92,156 +87,116 @@ func (kb dbKeybase) List() ([]Info, error) {
 	iter := kb.db.Iterator(nil, nil)
 	defer iter.Close()
 	for ; iter.Valid(); iter.Next() {
-		key := iter.Key()
-		if isPub(key) {
-			info, err := readInfo(iter.Value())
-			if err != nil {
-				return nil, err
-			}
-			res = append(res, info)
+		// key := iter.Key()
+		info, err := readInfo(iter.Value())
+		if err != nil {
+			return nil, err
 		}
+		res = append(res, info)
 	}
 	return res, nil
 }
 
 // Get returns the public information about one key.
 func (kb dbKeybase) Get(name string) (Info, error) {
-	bs := kb.db.Get(pubName(name))
+	bs := kb.db.Get(infoKey(name))
 	return readInfo(bs)
 }
 
 // Sign signs the msg with the named key.
 // It returns an error if the key doesn't exist or the decryption fails.
-// TODO: what if leddger fails ?
-func (kb dbKeybase) Sign(name, passphrase string, msg []byte) (sig crypto.Signature, pk crypto.PubKey, err error) {
-	var key crypto.PrivKey
-	armorStr := kb.db.Get(privName(name))
-	key, err = unarmorDecryptPrivKey(string(armorStr), passphrase)
+func (kb dbKeybase) Sign(name, passphrase string, msg []byte) (sig crypto.Signature, pub crypto.PubKey, err error) {
+	info, err := kb.Get(name)
 	if err != nil {
 		return
 	}
-
-	sig = key.Sign(msg)
-	pk = key.PubKey()
-	return
-}
-
-// Export decodes the private key with the current password, encrypts
-// it with a secure one-time password and generates an armored private key
-// that can be Imported by another dbKeybase.
-//
-// This is designed to copy from one device to another, or provide backups
-// during version updates.
-func (kb dbKeybase) Export(name, oldpass, transferpass string) ([]byte, error) {
-	armorStr := kb.db.Get(privName(name))
-	key, err := unarmorDecryptPrivKey(string(armorStr), oldpass)
+	priv, err := unarmorDecryptPrivKey(info.PrivKeyArmor, passphrase)
 	if err != nil {
-		return nil, err
+		return
 	}
+	sig = priv.Sign(msg)
+	pub = priv.PubKey()
+	return
+}
 
-	if transferpass == "" {
-		return key.Bytes(), nil
+func (kb dbKeybase) Export(name string) (armor string, err error) {
+	bz := kb.db.Get(infoKey(name))
+	if bz == nil {
+		return "", errors.New("No key to export with name " + name)
 	}
-	armorBytes := encryptArmorPrivKey(key, transferpass)
-	return []byte(armorBytes), nil
+	return armorInfoBytes(bz), nil
 }
 
-// Import accepts bytes generated by Export along with the same transferpass.
```

Comment:
Why remove comments?

Reply:
The old comment was wrong, and the API I think is self explanatory for now...

Reply:
Should we not by convention have a proper godoc comment on every exposed item ? Even if it's obvious?